public/app/features/explore/Table/TableContainer.tsx

```diff
@@ -85,33 +83,45 @@ export class TableContainer extends PureComponent<Props> {
       }
     }
 
-    const mainFrame = this.getMainFrame(dataFrames);
-    const subFrames = dataFrames?.filter((df) => df.meta?.custom?.parentRowIndex !== undefined);
+    const tableData: Array<{ main: DataFrame; sub?: DataFrame[] }> = [];
+    const mainFrames = this.getMainFrames(dataFrames).filter(
+      (frame: DataFrame | undefined): frame is DataFrame => !!frame && frame.length !== 0
+    );
+
+    mainFrames?.forEach((frame) => {
+      const subFrames =
+        dataFrames?.filter((df) => frame.refId === df.refId && df.meta?.custom?.parentRowIndex !== undefined) ||
+        undefined;
+      tableData.push({ main: frame, sub: subFrames });
+    });
+
+    const isNoData = tableData.length === 0;
```

Comment:
This does mean we will not show individual empty tables if queries are empty, only if all queries return 0 results. Not sure if that's correct or not.

Reply:
Hmm yeah I don't really mind anyway, either way it makes sense. I'd have a slight preference for combining any empty results into some kind of a warning for the user. Not in this PR though, but more of a longer term goal.

Reply:
I think it should be good for now 👍  

It seems to be a recurring theme of how we handle errors and empty results in Explore, we will need to look at this more holistically.